Adding Labels to Graph Blocks in Disassembly

PulakIIIT opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)


**Detailed description**

Note: I am open to suggestions from the community for design and implementation choices as this is my first time contributing to the project
<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->
- Added block entry address as the header_text of Disassembly Block. (Prior to this change the header_text element was never modified anywhere else) 
- Used "offset" color from theme to display it
- Right clicking on the entry address has the same context menu as that of first instruction
- Changed Graph Options Preferences Layout
- Current Result:
<img width="1245" alt="Screenshot 2020-11-20 at 1 59 43 PM" src="https://user-images.githubusercontent.com/56169176/99780246-390b3500-2b3c-11eb-9f2d-de191367fb25.png">

- Working on 
	- Fixing build errors

**Test plan (required)**
<!-- What steps should the reviewer take to test your pull request? Demonstrate that the code is solid. Example: The exact actions you made and their outcome. Add screenshots/videos if the pull request changes UI. This is your time to re-check that everything works and that you covered all the edge cases -->
- The block entry offset is visible on all platforms (I tested this on MacOS Catalina) and its right click context menu works as expected
- The rest of the instructions in the graph block have the correct right click context after enabling or disabling the block entry offset
- The graph menu option for hiding and showing the block entry offset works and reflects changes in real time
- The new layout of graph menu option layout looks well without any issues on different platforms + theme combinations

<!-- **Code formatting**
Make sure you ran astyle on your code before making the PR. Check our contribution guidelines here: https://cutter.re/docs/code.html -->

**Closing issues**
Closes #1646 
<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if such). -->

karliss commented:
Hi, my suggestion is making it an option in graph options page.

PulakIIIT commented:
Thanks for the suggestion. I have added it as a configuration option in Graphs pane in preferences.
<img width="876" alt="Screenshot 2020-11-21 at 2 35 07 PM" src="https://user-images.githubusercontent.com/56169176/99872286-cadf7480-2c06-11eb-9eb6-302a76b92224.png">
Checking and Unchecking the option instantly reflects the changes in the graph view

karliss commented:
I think it makes sense having it next to  "Show offsets" option and maybe elaborate label or tooltip for show offsets to help better distinguish the two options.

PulakIIIT commented:
I modified the graph options widget to make the layout and labels a bit more intuitive. I also added a tooltip for the graph block entry option label. 
<img width="900" alt="Screenshot 2020-11-22 at 12 37 34 PM" src="https://user-images.githubusercontent.com/56169176/99897480-d478e300-2cbf-11eb-82ab-a0e6d008224f.png">
Here is the tooltip
<img width="633" alt="Screenshot 2020-11-22 at 1 02 06 PM" src="https://user-images.githubusercontent.com/56169176/99897846-faec4d80-2cc2-11eb-8f0e-8dcf281d4f38.png">

karliss commented:
Once you feel it's ready remove the WIP/draft status and mark it as ready for review. Please also check the build errors in earlier commits. One of them happens in trunk, but one of them seems to be caused by your changes. Unfortunately travis builds aren't currently working, so it won't be possible to automatically verify the fix.

karliss reviewed (changes requested):
Two small changes and it will be good to merge. I wanted to ask you to rebase your branch so that it gets run with latest CI fixes but looks like you already did it. :+1:

PulakIIIT commented on src/common/Configuration.h:
```diff
@@ -186,6 +186,11 @@ class CUTTER_EXPORT Configuration : public QObject
     QPoint getGraphBlockSpacing();
     QPoint getGraphEdgeSpacing();
+    /**
+     * @brief Getter and setter for graph block entry offset
```
Apologies for the lazy documentation, I'll focus on writing better documentation in the future. Thanks for the constructive feedback.

karliss replied:
Don't worry about it too much. Sometimes there isn't anything to tell beside information already provided in function signature and common class documentation.

karliss commented on src/dialogs/preferences/GraphOptionsWidget.ui:
```diff
@@ -189,10 +231,6 @@
-  <tabstop>maxColsSpinBox</tabstop>
-  <tabstop>bitmapGraphScale</tabstop>
-  <tabstop>graphOffsetCheckBox</tabstop>
-  <tabstop>checkTransparent</tabstop>
   <tabstop>horizontalBlockSpacing</tabstop>
```
If you don't know what you what this does please read https://doc.qt.io/qt-5/designer-tab-order.html . The current order still seems weird to me.

ITAYC0HEN commented:
Thank you @PulakIIIT ! Very nice work!
Looking forward to see more contributions from you, this is a very nice solution you made here! :smiley_cat: 
Let us know if there is anything we can help with

PulakIIIT commented:
You're welcome @ITAYC0HEN. Thanks @karliss for helping me out at every step!
